spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/NacosDiscoveryProperties.java

```diff
@@ -279,6 +301,205 @@ public void init() throws Exception {
 		}
 	}
 
+	 class InetUtilsIPV6 implements Closeable {
+		private final ExecutorService executorService;
+
+		private final Log log = LogFactory.getLog(com.alibaba.cloud.nacos.NacosDiscoveryProperties.InetUtilsIPV6.class);
+
+		@Value("${spring.cloud.inetutils.default-hostname}")
+		private String defaultHostname = "localhost";
+
+		@Value("${spring.cloud.inetutils.default-ip-address}")
+		private String defaultIpAddress = "0:0:0:0:0:0:0:1";
+
+		@Value("${spring.cloud.inetutils.ignored-interfaces}")
+		private List<String> ignoredInterfaces = new ArrayList();
+
+		@Value("${spring.util.timeout.sec:${SPRING_UTIL_TIMEOUT_SEC:1}}")
+		private int timeoutSeconds = 1;
+
+		@Value("${spring.cloud.inetutils.use-only-site-local-interfaces}")
+		private boolean useOnlySiteLocalInterfaces = false;
+
+		@Value("${spring.cloud.inetutils.preferred-networks}")
+		private List<String> preferredNetworks = new ArrayList();
+
+		public InetUtilsIPV6() {
+			this.executorService = Executors.newSingleThreadExecutor((r) -> {
+				Thread thread = new Thread(r);
+				thread.setName("spring.cloud.alibaba.inetutilsIPV6");
+				thread.setDaemon(true);
+				return thread;
+			});
+		}
+
+		public void close() {
+			this.executorService.shutdown();
+		}
+
+		public InetUtils.HostInfo findFirstNonLoopbackHostInfo() {
+			InetAddress address = this.findFirstNonLoopbackAddress();
+			if (address != null) {
+				return this.convertAddress(address);
+			} else {
+				InetUtils.HostInfo hostInfo = new InetUtils.HostInfo();
+				hostInfo.setHostname(defaultHostname);
+				hostInfo.setIpAddress(defaultIpAddress);
+				return hostInfo;
+			}
+		}
+
+		public InetAddress findFirstNonLoopbackAddress() {
+			InetAddress result = null;
+
+			try {
+				int lowest = 2147483647;
+				Enumeration nics = NetworkInterface.getNetworkInterfaces();
+
+				label61:
+				while(true) {
```

Comment:
The logic of this paragraph is not very clear, it is a bit winding, and I wrote a lot of while loops.

Reply:
I will update the logic,and solve the problem above